### PR TITLE
fix: [PL-21099]: Connector details page tab view based on query params and added a popup when deleting referenced connector

### PR DESCRIPTION
**Pull request by avinashmadhwani02**
##### Summary:
Changes - 

1. Showing tab view in connector details page based on query param.
2. Showing popup while deleting a connector which is referenced by other entities.

Haven't written a test case for this flow, this needs to be merged today. Coverage is anyways not failing, will try to add test cases in the next PR

##### Screenshots:

Screenshot 2022-04-01 at 4 39 09 PM

You can use the following comments to re-trigger PR Checks

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress: `trigger cypress`
- Fix Prettier - `fix prettier`

### CODE REVIEW

File: src/modules/35-connectors/strings/strings.en.yaml [hunk 1]
```diff
@@ -784,3 +784,5 @@ ceGcp:
     heading: '{{ $.common.labelTestConnection }}'
     step1: 'Verifying GCP connector setup'
     error: 'Could not verify connectivity'
+connectorReferenceText: 'cannot be deleted as it is being referenced by other entities. To see the list of entities referencing to this connector,'
```
Comment (sunnykesh-harness) [inline src/modules/35-connectors/strings/strings.en.yaml:787]: It should be either **can not** or **can't** . Please confirm with doc team.

File: src/modules/35-connectors/pages/connectors/views/ConnectorsListView.tsx [hunk 2]
```diff
@@ -243,6 +245,37 @@ const RenderColumnMenu: Renderer<CellProps<ConnectorResponse>> = ({ row, column
     )
   }
 
+  const { openDialog: openReferenceErrorDialog } = useConfirmationDialog({
+    contentText: (
+      <Text inline>
```
Comment (sunnykesh-harness) [inline src/modules/35-connectors/pages/connectors/views/ConnectorsListView.tsx:250]: Is nested `Text` throwing any console error/warning?

**Review by sunnykesh-harness — APPROVED** [review]
1. The modal showing referenced by error should be consistent. It is different than when a referenced service deleting fails.
2. As discussed please test for gitSync.

File: src/modules/35-connectors/strings/strings.en.yaml (hunk 1 above)
Comment (avinashmadhwani02) [reply src/modules/35-connectors/strings/strings.en.yaml:787]: I think @sunnykesh-harness `cannot` is also fine, it's given by Kapil and we are using the same in the title also.

File: src/modules/35-connectors/pages/connectors/views/ConnectorsListView.tsx (hunk 2 above)
Comment (avinashmadhwani02) [reply src/modules/35-connectors/pages/connectors/views/ConnectorsListView.tsx:250]: Any reason for error/warning?

File: src/modules/10-common/RouteDefinitions.ts [hunk 3]
```diff
@@ -240,9 +241,14 @@ const routes = {
...
       connectorId,
-      module
-    }: Partial<ProjectPathProps & ModulePathParams & ConnectorPathProps>) => {
-      const path = `resources/connectors/${connectorId}`
+      module,
+      view
+    }: Partial<ProjectPathProps & ModulePathParams & ConnectorPathProps & ConnectorDetailsQueryParams>) => {
+      let path = `resources/connectors/${connectorId}`
+      if (view) {
+        path = `resources/connectors/${connectorId}?view=${view}`
```
Comment (abhinavrastogi-harness) [inline src/modules/10-common/RouteDefinitions.ts:249]: query params should NOT be added to the route.
pass them as a second param of `history.push` instead.

File: src/modules/35-connectors/pages/connectors/ConnectorDetailsPage/ConnectorDetailsPage.tsx [hunk 4]
```diff
@@ -240,14 +248,25 @@ const ConnectorDetailsPage: React.FC<ConnectorDetailsPageProps> = props => {
...
+                    className={cx(css.tags, css.small, { [css.active]: activeCategory === item })}
+                    onClick={() => {
+                      history.push(
+                        routes.toConnectorDetails({
+                          accountId,
+                          orgIdentifier,
+                          projectIdentifier,
+                          module,
+                          connectorId,
+                          view: item
```
Comment (abhinavrastogi-harness) [inline src/modules/35-connectors/pages/connectors/ConnectorDetailsPage/ConnectorDetailsPage.tsx:263]: move this to the second param of `history.push(route.toConnectorDetails(...), {search: '?view=xyz'})`

File: src/modules/35-connectors/pages/connectors/views/ConnectorsListView.tsx (hunk 2 above)
Comment (abhinavrastogi-harness) [reply src/modules/35-connectors/pages/connectors/views/ConnectorsListView.tsx:250]: +1
Text should not be nested

File: src/modules/35-connectors/pages/connectors/views/ConnectorsListView.tsx [hunk 5]
```diff
@@ -21,10 +21,11 @@ import {
 import { FontVariation, Color } from '@harness/design-system'
 import type { CellProps, Renderer, Column } from 'react-table'
 import { Menu, Classes, Position, Intent, TextArea, Tooltip } from '@blueprintjs/core'
-import { useParams, useHistory } from 'react-router-dom'
+import { useParams, useHistory, Link } from 'react-router-dom'
 import ReactTimeago from 'react-timeago'
 import classNames from 'classnames'
 import { pick } from 'lodash-es'
+import defaultTo from 'lodash/defaultTo'
```
Comment (abhinavrastogi-harness) [inline src/modules/35-connectors/pages/connectors/views/ConnectorsListView.tsx:28]: import from `lodash-es`

File: src/modules/35-connectors/pages/connectors/utils/ConnectorHelper.tsx [hunk 6]
```diff
@@ -217,6 +217,12 @@ export const getConnectorIconByType = (type: string): IconName => {
   }
 }
 
+export enum ConnectorDetailsView {
+  'overview' = 'overview',
+  'refrencedBy' = 'refrencedBy',
```
Comment (abhinavrastogi-harness) [inline src/modules/35-connectors/pages/connectors/utils/ConnectorHelper.tsx:222]: spelling mistake

File: src/modules/35-connectors/strings/strings.en.yaml (hunk 1 above)
Comment (abhinavrastogi-harness) [reply src/modules/35-connectors/strings/strings.en.yaml:787]: `referencing this connector` instead of `referencing to this connector`

**Review by abhinavrastogi-harness — CHANGES_REQUESTED** [review]

File: src/modules/35-connectors/pages/connectors/views/ConnectorsListView.tsx [hunk 7]
```diff
@@ -21,10 +21,11 @@ import {
 import { FontVariation, Color } from '@harness/design-system'
 import type { CellProps, Renderer, Column } from 'react-table'
 import { Menu, Classes, Position, Intent, TextArea, Tooltip } from '@blueprintjs/core'
-import { useParams, useHistory } from 'react-router-dom'
+import { useParams, useHistory, Link } from 'react-router-dom'
 import ReactTimeago from 'react-timeago'
 import classNames from 'classnames'
 import { pick } from 'lodash-es'
+import defaultTo from 'lodash-es/defaultTo'
```
Comment (abhinavrastogi-harness) [inline src/modules/35-connectors/pages/connectors/views/ConnectorsListView.tsx:28]: combine this with above import..
`import { pick, defaultTo } from 'lodash-es'`

**Review by abhinavrastogi-harness — APPROVED** [review]

File: src/modules/35-connectors/pages/connectors/ConnectorDetailsPage/ConnectorDetailsPage.tsx [hunk 8]
```diff
@@ -84,6 +92,12 @@ const ConnectorDetailsPage: React.FC<ConnectorDetailsPageProps> = props => {
     }
   }, [connectorData, loading])
 
+  useEffect(() => {
+    if (view) {
+      setActiveCategory(view)
+    }
+  }, [view])
```
[the remark this answers is not in the document]
Comment (avinashmadhwani02) [reply src/modules/35-connectors/pages/connectors/ConnectorDetailsPage/ConnectorDetailsPage.tsx:99]: Yes we do

**Review by VikasMaddukuriHarness — APPROVED** [review]

**Pull request merged**